Userland/Libraries/LibCompress/BrotliDictionary.cpp

```diff
@@ -0,0 +1,7902 @@
+    0x77, 0x69, 0x64, 0x74, 0x68, 0x3d, 0x22, 0x32, 0x6c, 0x61, 0x7a, 0x79, 0x6c, 0x6f, 0x61, 0x64,
+    0x6e, 0x6f, 0x76, 0x65, 0x6d, 0x62, 0x65, 0x72, 0x75, 0x73, 0x65, 0x64, 0x20, 0x69, 0x6e, 0x20,
+    0x68, 0x65, 0x69, 0x67, 0x68, 0x74, 0x3d, 0x22, 0x63, 0x72, 0x69, 0x70, 0x74, 0x22, 0x3e, 0x0a,
+    0x26, 0x6e, 0x62, 0x73, 0x70, 0x3b, 0x3c, 0x2f, 0x3c, 0x74, 0x72, 0x3e, 0x3c, 0x74, 0x64, 0x20,
+    0x68, 0x65, 0x69, 0x67, 0x68, 0x74, 0x3a, 0x32, 0x2f, 0x70, 0x72, 0x6f, 0x64, 0x75, 0x63, 0x74,
+    0x63, 0x6f, 0x75, 0x6e, 0x74, 0x72, 0x79, 0x20, 0x69, 0x6e, 0x63, 0x6c, 0x75, 0x64, 0x65, 0x20,
+    0x66, 0x6f, 0x6f, 0x74, 0x65, 0x72, 0x22, 0x20, 0x26, 0x6c, 0x74, 0x3b, 0x21, 0x2d, 0x2d, 0x20,
+    0x74, 0x69, 0x74, 0x6c, 0x65, 0x22, 0x3e, 0x3c, 0x2f, 0x6a, 0x71, 0x75, 0x65, 0x72, 0x79, 0x2e,
+    0x3c, 0x2f, 0x66, 0x6f, 0x72, 0x6d, 0x3e, 0x0a, 0x28, 0xe7, 0xae, 0x80, 0xe4, 0xbd, 0x93, 0x29,
+    0x28, 0xe7, 0xb9, 0x81, 0xe9, 0xab, 0x94, 0x29, 0x68, 0x72, 0x76, 0x61, 0x74, 0x73, 0x6b, 0x69,
+    0x69, 0x74, 0x61, 0x6c, 0x69, 0x61, 0x6e, 0x6f, 0x72, 0x6f, 0x6d, 0xc3, 0xa2, 0x6e, 0xc4, 0x83,
+    0x74, 0xc3, 0xbc, 0x72, 0x6b, 0xc3, 0xa7, 0x65, 0xd8, 0xa7, 0xd8, 0xb1, 0xd8, 0xaf, 0xd9, 0x88,
+    0x74, 0x61, 0x6d, 0x62, 0x69, 0xc3, 0xa9, 0x6e, 0x6e, 0x6f, 0x74, 0x69, 0x63, 0x69, 0x61, 0x73,
+    0x6d, 0x65, 0x6e, 0x73, 0x61, 0x6a, 0x65, 0x73, 0x70, 0x65, 0x72, 0x73, 0x6f, 0x6e, 0x61, 0x73,
+    0x64, 0x65, 0x72, 0x65, 0x63, 0x68, 0x6f, 0x73, 0x6e, 0x61, 0x63, 0x69, 0x6f, 0x6e, 0x61, 0x6c,
+    0x73, 0x65, 0x72, 0x76, 0x69, 0x63, 0x69, 0x6f, 0x63, 0x6f, 0x6e, 0x74, 0x61, 0x63, 0x74, 0x6f,
+    0x75, 0x73, 0x75, 0x61, 0x72, 0x69, 0x6f, 0x73, 0x70, 0x72, 0x6f, 0x67, 0x72, 0x61, 0x6d, 0x61,
+    0x67, 0x6f, 0x62, 0x69, 0x65, 0x72, 0x6e, 0x6f, 0x65, 0x6d, 0x70, 0x72, 0x65, 0x73, 0x61, 0x73,
+    0x61, 0x6e, 0x75, 0x6e, 0x63, 0x69, 0x6f, 0x73, 0x76, 0x61, 0x6c, 0x65, 0x6e, 0x63, 0x69, 0x61,
+    0x63, 0x6f, 0x6c, 0x6f, 0x6d, 0x62, 0x69, 0x61, 0x64, 0x65, 0x73, 0x70, 0x75, 0xc3, 0xa9, 0x73,
+    0x64, 0x65, 0x70, 0x6f, 0x72, 0x74, 0x65, 0x73, 0x70, 0x72, 0x6f, 0x79, 0x65, 0x63, 0x74, 0x6f,
+    0x70, 0x72, 0x6f, 0x64, 0x75, 0x63, 0x74, 0x6f, 0x70, 0xc3, 0xba, 0x62, 0x6c, 0x69, 0x63, 0x6f,
+    0x6e, 0x6f, 0x73, 0x6f, 0x74, 0x72, 0x6f, 0x73, 0x68, 0x69, 0x73, 0x74, 0x6f, 0x72, 0x69, 0x61,
+    0x70, 0x72, 0x65, 0x73, 0x65, 0x6e, 0x74, 0x65, 0x6d, 0x69, 0x6c, 0x6c, 0x6f, 0x6e, 0x65, 0x73,
+    0x6d, 0x65, 0x64, 0x69, 0x61, 0x6e, 0x74, 0x65, 0x70, 0x72, 0x65, 0x67, 0x75, 0x6e, 0x74, 0x61,
+    0x61, 0x6e, 0x74, 0x65, 0x72, 0x69, 0x6f, 0x72, 0x72, 0x65, 0x63, 0x75, 0x72, 0x73, 0x6f, 0x73,
+    0x70, 0x72, 0x6f, 0x62, 0x6c, 0x65, 0x6d, 0x61, 0x73, 0x61, 0x6e, 0x74, 0x69, 0x61, 0x67, 0x6f,
+    0x6e, 0x75, 0x65, 0x73, 0x74, 0x72, 0x6f, 0x73, 0x6f, 0x70, 0x69, 0x6e, 0x69, 0xc3, 0xb3, 0x6e,
+    0x69, 0x6d, 0x70, 0x72, 0x69, 0x6d, 0x69, 0x72, 0x6d, 0x69, 0x65, 0x6e, 0x74
```

Comment:
This can fail, for example by going to `cloudflare.com`:
```
28.845 RequestServer(44:44): ASSERTION FAILED: 4 <= length && length <= 24
../../Userland/Libraries/LibCompress/BrotliDictionary.cpp:7866
28.848 [#0 RequestServer(44:44)]: Terminating RequestServer(44) due to signal 6
28.848 [#0 FinalizerTask(4:4)]: Backtrace:
0x00000000deadc0de  Kernel::Processor::switch_context(Kernel::Thread*&, Kernel::Thread*&) + 0x383
28.848 [#0 FinalizerTask(4:4)]: Generating coredump for pid: 44
28.879 CrashDaemon(9:9): New coredump file: /tmp/coredump/RequestServer_44_1649194443
29.005 WebContent(46:46): Parse error! [handle_initial @ ../../Userland/Libraries/LibWeb/HTML/Parser/HTMLParser.cpp:469]
29.013 CrashReporter(45:45): Started thread "", tid = 47
29.382 CrashReporter(45:47): Generating backtrace took 357 ms
29.382 CrashReporter(45:47): --- Backtrace for thread #0 (TID 44) ---
29.382 CrashReporter(45:47): 0x0000001b995f702b: [libsystem.so] syscall2 +0xb (syscall.cpp:25 => syscall.cpp:24)
29.382 CrashReporter(45:47): 0x0000000a57877ba7: [libc.so] abort +0x27 (stdlib.cpp:223)
29.382 CrashReporter(45:47): 0x0000000a5787cb3e: [libc.so] __assertion_failed +0x7e (assert.cpp:33)
29.382 CrashReporter(45:47): 0x0000000460cc795b: [libcompress.so.serenity] Compress::BrotliDictionary::lookup_word(unsigned long, unsigned long) +0x20b (BrotliDictionary.cpp:7866)
29.382 CrashReporter(45:47): 0x0000000460cc52b3: [libcompress.so.serenity] Compress::BrotliDecompressionStream::read(AK::Span<unsigned char>) +0x15f3 (Brotli.cpp:853)
29.386 CrashReporter(45:47): 0x000000181b0ec289: [libcore.so.serenity] Core::Stream::Stream::read_all() +0x69 (Stream.cpp:57)
29.386 CrashReporter(45:47): 0x000000183ddcc0f4: [libhttp.so.serenity] HTTP::Job::finish_up() [clone .localalias] +0xff4 (Job.cpp:86 => Job.cpp:613)
29.386 CrashReporter(45:47): 0x000000183ddcee4f: [libhttp.so.serenity] HTTP::Job::on_socket_connected()::{lambda()#2}::operator()() const +0x251f (Job.cpp:323)
29.386 CrashReporter(45:47): 0x000000183ddc95c3: [libhttp.so.serenity] AK::Function<void ()>::CallableWrapper<HTTP::Job::register_on_ready_to_read(AK::Function<void ()>)::{lambda()#1}>::call() +0x63 (Function.h:91)
29.386 CrashReporter(45:47): 0x000000183ddc9880: [libhttp.so.serenity] AK::Function<void ()>::CallableWrapper<HTTP::Job::register_on_ready_to_read(AK::Function<void ()>)::{lambda()#1}::operator()() const::{lambda()#2}>::call() +0x50 (Function.h:91)
29.390 CrashReporter(45:47): 0x000000181b0e180c: [libcore.so.serenity] AK::Function<void ()>::CallableWrapper<Core::Object::deferred_invoke(AK::Function<void ()>)::{lambda()#1}>::call() +0x23c (Function.h:91)
29.390 CrashReporter(45:47): 0x000000181b0caa6a: [libcore.so.serenity] Core::EventLoop::pump(Core::EventLoop::WaitMode) [clone .localalias] +0x28a (Function.h:91)
29.390 CrashReporter(45:47): 0x000000181b0cb229: [libcore.so.serenity] Core::EventLoop::exec() +0x119 (EventLoop.cpp:438)
29.390 CrashReporter(45:47): 0x00000006bff5b5c1: [/bin/RequestServer] serenity_main(Main::Arguments) +0x1b1 (main.cpp:51)
29.390 CrashReporter(45:47): 0x00000006bff2e915: [/bin/RequestServer] main +0x95 (Main.cpp:39)
29.394 CrashReporter(45:47): 0x00000006bff2eb5f: [/bin/RequestServer] _entry +0x6f (crt0.cpp:49)
```
I got a fuzzed example file:
[crash-cfea2e3d833661611ae605b1a62e68a23d110dae.txt](https://github.com/SerenityOS/serenity/files/8422042/crash-cfea2e3d833661611ae605b1a62e68a23d110dae.txt)